apps/nextjs/src/app/legal/[slug]/page.tsx

```diff
@@ -0,0 +1,18 @@
+import { fetchPolicyDocument } from "cms/data/fetchPolicyDocument";
+import { redirect } from "next/navigation";
+
+import LegalContent from "./legal";
+
+export default async function PolicyContentPage({
+  params,
+}: {
+  params: { slug: string };
+}) {
+  const pageData = await fetchPolicyDocument({ slug: params.slug });
+
+  if (!pageData) {
+    redirect("/not-found");
+  }
```

Comment:
I think we can return a not found response in a server component rather than redirecting to one?